keanu-project/src/test/java/io/improbable/keanu/algorithms/mcmc/proposal/AcceptanceRateTrackerTest.java

```diff
@@ -0,0 +1,100 @@
+package io.improbable.keanu.algorithms.mcmc.proposal;
+
+import com.google.common.collect.ImmutableSet;
+import io.improbable.keanu.vertices.Vertex;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.Mockito.mock;
+
+public class AcceptanceRateTrackerTest {
+
+    private Vertex vertex1 = mock(Vertex.class);
+    private Vertex vertex2 = mock(Vertex.class);
+    Proposal proposal;
+    private AcceptanceRateTracker acceptanceRateTracker;
+
+    @Before
+    public void createTracker() throws Exception {
+        acceptanceRateTracker = new AcceptanceRateTracker();
+    }
+
+    @Test
+    public void theAcceptanceRateForAnUnrecognisedSetOfVerticesIsNaN() {
```

Comment:
Why not throw an exception if unrecognised vertices?

Reply:
You don't have to throw a checked exception. An IllegalArgument exception or something like that would make sense. It's very python to just return NaN and let the user figure it out.

Reply:
OK, I've implemented it as an `IllegalStateException`.

Whether or not it's a `RuntimeException` isn't the issue though - I still need to handle it in my `log.trace` logic so that it doesn't throw when logging the first call to `onProposalApplied`